### PR TITLE
Fixes #4229 - App Crashes while trying to nominate an item for deletion, When User is not logged in

### LINKED ISSUE #4229: App Crashes while trying to nominate an item for deletion, When User is not logged in.
**Summary:** 
When the user is not logged in, App Crashes after giving the reason while nominating the item for deletion.

**Steps to reproduce:** 

1. Open the app without logging in.
2. Click on any Media to open media detail.
3. Now try to nominate the media for deletion and click "ok" after giving the reason.

What did you expect the app to do, and what did you see instead?
The app should redirect the user to the login screen.

**System logs:**

> E/AndroidRuntime: FATAL EXCEPTION: main

    Process: fr.free.nrw.commons.beta, PID: 18068
    android.util.AndroidRuntimeException: Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?
        at android.app.ContextImpl.startActivity(ContextImpl.java:952)
        at android.app.ContextImpl.startActivity(ContextImpl.java:928)
        at android.content.ContextWrapper.startActivity(ContextWrapper.java:383)
        at fr.free.nrw.commons.auth.LoginActivity.startYourself(LoginActivity.java:459)
        at fr.free.nrw.commons.auth.SessionManager.forceLogin(SessionManager.java:121)
        at fr.free.nrw.commons.delete.ReasonBuilder.checkAccount(ReasonBuilder.java:96)
        at fr.free.nrw.commons.delete.ReasonBuilder.fetchArticleNumber(ReasonBuilder.java:66)
        at fr.free.nrw.commons.delete.ReasonBuilder.getReason(ReasonBuilder.java:51)
        at fr.free.nrw.commons.media.MediaDetailFragment.onDeleteClickeddialogtext(MediaDetailFragment.java:769)
        at fr.free.nrw.commons.media.MediaDetailFragment.lambda$onDeleteButtonClicked$4$MediaDetailFragment(MediaDetailFragment.java:717)
        at fr.free.nrw.commons.media.-$$Lambda$MediaDetailFragment$6CeKf5Kd-vAslDfXw_b_q2Gutuc.onClick(Unknown Source:4)
        at com.android.internal.app.AlertController$ButtonHandler.handleMessage(AlertController.java:177)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)

**Device and Android version:** 
Device: Pixel 3 emulator, Android 10
 
**Commons app version:** 
Latest Master

**Would you like to work on the issue?**
Yes :-)

### PR DESCRIPTION
**Pull request by Pratham2305**
**Description (required)**

Fixes #4229 

What changes did you make and why?
Added a condition to check the user is logged in or not, if not logged in then simply hide the delete button. As the feature is of no use for the non-logged-in users.

**Tests performed (required)**
Tested betaDebug on Pixel 3 with API level 29.

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java [hunk 1]
```diff
@@ -258,6 +263,10 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
             authorLayout.setVisibility(GONE);
         }
 
+        if(applicationKvStore.getBoolean("login_skipped") == true){
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java:266]: Please remove `== true`.
Shorter and easier to read :-)

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
I can reproduce the crash in latest master.
With this PR, the button is not visible.
Conclusion: This PR fixes the bug :-)

Just remove the `== true` and I will merge :-)

File: app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java (hunk 1 above)
Comment (Pratham2305) [reply app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java]: Done!

**Comment by nicolas-raoul** [comment]
Thanks Pratham!

**Pull request merged**